src/reload.lisp

```diff
@@ -45,8 +45,11 @@
                 "/sw/lib/libcrypto.dylib"        ;; Fink
                 "/usr/local/opt/openssl/lib/libcrypto.dylib" ;; Homebrew
                 "/usr/local/lib/libcrypto.dylib" ;; personalized install
-                "libcrypto.dylib"                ;; default system libcrypto, which may have insufficient crypto
-                "/usr/lib/libcrypto.dylib"))
+                ;; system-provided libraries
+                "/usr/lib/libcrypto.44.dylib"
+                "/usr/lib/libcrypto.42.dylib"
+                "/usr/lib/libcrypto.41.dylib"
+                "/usr/lib/libcrypto.35.dylib"))
```

Comment:
Do we want to remove both "libcrypto.dylib" and /usr/lib/libcrypto.dylib"? Maybe just put them at the end of the list?

Also, do we want the MacPorts, Fink and Homebrew paths to be tried before the new versioned paths? (CFFI tries the paths in the specified order)

Reply:
It depends. There is no problem with trying those other paths first, and it might help people who want to install their own version of openssl. 

The issue is mostly that loading "/usr/lib/libcrypto.dylib" or just "libcrypto.dylib" causes the crash, so that one shouldn't be in the list. It seems that Apple decided it's a bad idea to load it like that in macOS 10.15. They made it print a warning in macOS 10.15, but I guess nobody noticed because of homebrew. In macOS 11.0, they made this warning fatal, and I only managed to trigger it because homebrew has moved their default prefix to`/opt/homeberw` for arm64.

Reply:
We could consider adding "/usr/lib/libcrypto.dylib" back for compatibility with very old macOS systems.

Reply:
Any chance we can add `/opt/homebrew/opt/openssl/lib ;; Homebrew Arm64` to this list?  That would take care of us darwin arm64 people.

Reply:
Sorry, It was late last night when I made that comment.  I should have just submitted a PR.  I submitted https://github.com/cl-plus-ssl/cl-plus-ssl/pull/119 to address it.